src/Components/DraggableResizer/DraggableResizer.tsx

```diff
@@ -126,27 +126,58 @@ export const DraggableResizerRoot: FunctionComponent<DraggableResizerProps> = ({
     }
 
     const newPos = mouseRelativePos / containerSize
-    const newPositions = [...handlePositions]
+    createNewPositions(newPos)
+  }
+
+  const calculateCollapsePosition = (direction: number, dragIndex: number) => {
+    const totalDrag = handlePositions.length - 1
+    const directionIndex = direction === 0 ? dragIndex : dragIndex + 2
+    const currentPosition = handlePositions[dragIndex]
+    let collapsedPosition = directionIndex * 0.25
+
+    if (dragIndex > 0 && dragIndex < totalDrag) {
+      const pDragIndex = dragIndex - 1
+      const pCollapsedPosition = directionIndex * 0.25
+      const pCurrentPosition = handlePositions[pDragIndex]
+
+      collapsedPosition =
+        pCurrentPosition === pCollapsedPosition
+          ? pCollapsedPosition
+          : collapsedPosition
+    }
+
+    return currentPosition === collapsedPosition
+      ? startPosition[dragIndex]
+      : collapsedPosition
+  }
 
+  const onCollapseButtonClick = (direction: number, dragIndex: number) => {
+    const pos = calculateCollapsePosition(direction, dragIndex)
+    createNewPositions(pos, dragIndex)
+  }
+
+  const createNewPositions = (newPos: number, nextDragIndex?: number) => {
+    const di = nextDragIndex !== undefined ? nextDragIndex : dragIndex
```

Comment:
is the intent to store `nextDragIndex`? We should name the variable to that or something more verbose.

Reply:
No, it t just exists if its passed to be checked